File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.ts
function countCommas(n: number): number {
    if (n <= 999) return 0;
    let total = 0;
    let start = 1000;
    while (start <= n) {
        total += n - start + 1;
        start *= 1000;
    }
    return total;
}

File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.rs
impl Solution {
    pub fn count_commas(n: i64) -> i64 {
        if n <= 999 { return 0; }
        let mut total = 0;
        let mut start = 1000;
        while start <= n {
            total += n - start + 1;
            start *= 1000;
        }
        total
    }
}